$NetBSD: patch-an,v 1.2 2008/04/08 01:44:47 bjs Exp $

This driver is buggy on NetBSD right now.  This is likely [in part] due to
problems with our audio driver, but there could very well be problems
here, too.  I welcome guidance/suggestions/comments on how to improve
this driver, which was written by <jakemsr@openbsd.org>.  For NetBSD
questions, though, it's probably best if you contact the maintainer of
this package and/or another NetBSD developer.

--- drivers/sun/sun_driver.c.orig	2008-03-27 17:29:08.000000000 -0400
+++ drivers/sun/sun_driver.c
@@ -26,13 +26,6 @@
 
 #include <config.h>
 
-#ifndef _REENTRANT
-#define _REENTRANT
-#endif
-#ifndef _THREAD_SAFE
-#define _THREAD_SAFE
-#endif
-
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
@@ -59,6 +52,12 @@
 
 #include "sun_driver.h"
 
+#ifdef __NetBSD__
+#define AUMODE_PLAY_FLAGS	(AUMODE_PLAY | AUMODE_PLAY_ALL)
+#else
+#define AUMODE_PLAY_FLAGS	AUMODE_PLAY
+#endif
+
 
 #define SUN_DRIVER_N_PARAMS	11
 const static jack_driver_param_desc_t sun_params[SUN_DRIVER_N_PARAMS] = {
@@ -736,7 +735,7 @@ sun_driver_set_parameters (sun_driver_t 
 
 	if (infd == outfd)
 	{
-		audio_if_in.mode = AUMODE_PLAY | AUMODE_RECORD;
+		audio_if_in.mode = AUMODE_PLAY_FLAGS | AUMODE_RECORD;
 	}
 	else
 	{
@@ -744,7 +743,7 @@ sun_driver_set_parameters (sun_driver_t 
 			audio_if_in.mode = AUMODE_RECORD;
 
 		if (outfd > 0)
-			audio_if_out.mode = AUMODE_PLAY;
+			audio_if_out.mode = AUMODE_PLAY_FLAGS;
 	}
 
 	if (infd > 0)
